src/cookieconsent.js

```diff
@@ -0,0 +1,1306 @@
+        if (typeof v === 'object' && v !== null) {
+          dest[prop] = dest[prop] || {};
+          this.deepExtend(dest[prop], v);
+        } else {
+          dest[prop] = v;
+        }
+      }
+      return dest;
+    },
+
+    // only used for throttling the 'mousemove' event (used for animating the revoke button when `animateRevokable` is true)
+    throttle: function (callback, limit) {
+      var wait = false;
+      return function () {
+        if (!wait) {
+          callback.apply(this, arguments);
+          wait = true;
+          setTimeout(function () {
+            wait = false;
+          }, limit);
+        }
+      }
+    },
+
+    // only used for hashing json objects (used for hash mapping palette objects, used when custom colours are passed through javascript)
+    hash: function (str) {
+      var hash = 0, i, chr, len;
+      if (str.length === 0) return hash;
+      for (i = 0, len = str.length; i < len; i++) {
+        chr   = str.charCodeAt(i);
+        hash  = ((hash << 5) - hash) + chr;
+        hash |= 0;
+      }
+      return hash;
+    },
+
+    validateHex: function(hex) {
+      hex = hex.substr(1);
+      if (hex.length < 6) {
+        hex = hex[0]+hex[0]+hex[1]+hex[1]+hex[2]+hex[2];
+      }
+      return hex;
+    },
+
+    // used to get text colors if not set
+    getContrast: function (hex){
+      hex = util.validateHex(hex);
+      var r = parseInt(hex.substr(0,2),16);
+      var g = parseInt(hex.substr(2,2),16);
+      var b = parseInt(hex.substr(4,2),16);
+      var yiq = ((r*299)+(g*587)+(b*114))/1000;
+      return (yiq >= 128) ? '#000' : '#fff';
+    },
+
+    // used to change color on highlight
+    getLuminance: function(hex) {
+      hex = util.validateHex(hex);
+      if (hex=='000000') return '#222'; //for black buttons
+      lum = 0.2;
+      var rgb = "#", c, i;
+      for (i = 0; i < 3; i++) {
+        c = parseInt(hex.substr(i*2,2), 16);
+        c = Math.round(Math.min(Math.max(0, c + (c * lum)), 255)).toString(16);
+        rgb += ("00"+c).substr(c.length);
+      }
+      return rgb;
+    },
+
+    // getting the directory of javascript file
+    getDirectory: function() {
+      sc = document.getElementsByTagName("script");
+      for(i = 0; i < sc.length; i++) {
+        s = sc.item(i);
+        if(s.src && s.src.match(/cookieconsent.min.js$/))
+        { return s.src.slice(0,-'cookieconsent.min.js'.length); }
+      }
+      return '';
+    },
+
+    isMobile: function() {
+      if( /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent) ) return true;
+      return false;
+    }
+
+  };
+
+  // valid cookie values
+  cc.status = {deny: 'deny', allow: 'allow', dismiss: 'dismiss'};
+
+  // is true if the browser supports css transitions
+  cc.hasTransition = (function () {
+    var style = document.documentElement.style;
+    var trans = ['t','OT','msT','MozT','KhtmlT','WebkitT'];
+    for (var i = 0, l = trans.length; i < l; ++i) {
```

Comment:
I am wrong. This is correct. I apologies @alexmorleyfinch for slandering your good name